### PR TITLE
Add configs for cloud vision APIs.

### PR DESCRIPTION
**Pull request by jmuk**
- gapic.yaml is generated by toolkit
- others are hand-written

### CODE REVIEW

File: google/cloud/vision/v1/vision_gapic.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    flattening:
+      groups:
+      - parameters:
+        - requests
+    required_fields:
+    - requests
+    request_object_method: false
+    retry_codes_name: non_idempotent
+    retry_params_name: default
+    timeout_millis: 30000
```
Comment (garrettjonesgoogle) [inline google/cloud/vision/v1/vision_gapic.yaml]: make 60000 (do we need to update the config generator?)

Comment (jmuk) [inline google/cloud/vision/v1/vision_gapic.yaml]: Fixed manually. The issue for config generator seems filed as https://github.com/googleapis/toolkit/issues/263

**Comment by garrettjonesgoogle** [comment]
LGTM

File: google/cloud/vision/v1/vision.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+documentation:
+  summary:
+    'Integrates Google Vision features, including image labeling, face, logo,
+    and landmark detection, optical character recognition (OCR), and detection
+    of explicit content, into applications.'
+
+authentication:
+  rules:
+    - selector: '*'
+      allow_without_credential: true
```
Comment (bjwatson) [inline google/cloud/vision/v1/vision.yaml]: Logging and pubsub don't have `allow_without_credential`, so it might not be needed.

Comment (garrettjonesgoogle) [inline google/cloud/vision/v1/vision.yaml]: But wait, maybe this says something interesting - does it mean someone can call the API with empty credentials?

File: google/cloud/vision/v1/vision_gapic.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  methods:
+  - name: BatchAnnotateImages
+    flattening:
+      groups:
+      - parameters:
+        - requests
+    required_fields:
+    - requests
+    request_object_method: false
+    retry_codes_name: non_idempotent
```
Comment (bjwatson) [inline google/cloud/vision/v1/vision_gapic.yaml]: Seems like this could be idempotent

File: google/cloud/vision/v1/vision.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      oauth:
+        canonical_scopes: https://www.googleapis.com/auth/cloud-platform
+
+context:
```
Comment (bjwatson) [inline google/cloud/vision/v1/vision.yaml]: As @jmuk and I discussed offline, we don't seem to need this section (pubsub and logging don't have it).

File: google/cloud/vision/v1/vision.yaml (hunk 2 above)
Comment (jmuk) [inline google/cloud/vision/v1/vision.yaml]: Hm... Removed.

I actually brought this line from internal, but at least cloud-vision API site says about the authentication (https://cloud.google.com/vision/docs/auth-template/cloud-api-auth), so this is probably not the case of this API at least for now.

But I think gapic-generated code does not consider APIs which don't require credentials, so this is an interesting topic.

File: google/cloud/vision/v1/vision_gapic.yaml (hunk 3 above)
Comment (jmuk) [inline google/cloud/vision/v1/vision_gapic.yaml]: fixed

File: google/cloud/vision/v1/vision.yaml (hunk 4 above)
Comment (jmuk) [inline google/cloud/vision/v1/vision.yaml]: removed

File: google/cloud/vision/v1/vision.yaml (hunk 2 above)
Comment (bjwatson) [inline google/cloud/vision/v1/vision.yaml]: It seems that way: https://github.com/googleapis/googleapis/blob/master/google/api/auth.proto#L70

Do you think there's something we could do with this information in our generators?

**Comment by bjwatson** [comment]
LGTM

File: google/cloud/vision/v1/vision.yaml (hunk 2 above)
Comment (jmuk) [inline google/cloud/vision/v1/vision.yaml]: yeah, generated code assumes to pass list of scopes by default, and gax-python (and other gax libraries I think) will create a channel-credential based on the scopes, which could be unnecessary.

Comment (bjwatson) [inline google/cloud/vision/v1/vision.yaml]: IIUC this is an opportunity for efficiency improvements, but does not affect functionality, right?

**Pull request merged**